src/traits.rs

```diff
@@ -204,3 +204,43 @@ pub trait Xgcd<T = Self> {
     /// `x`, and `y` s.t. `gcd(a,b) = a*x + b*y`.
     fn xgcd(&self, other: T) -> Self::Output;
 }
+
+/// Implemented by [`MatZ`](crate::integer::MatZ) and further matrix types whose
+/// columns or rows are to be conveniently swapped.
+pub trait MatrixSwaps {
+    /// Swaps two columns of the specified matrix.
+    ///
+    /// Parameters:
+    /// - `col0`: specifies the first column which is swapped with the second one
+    /// - `col1`: specifies the second column which is swapped with the first one
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// Otherwise, a [`MathError`] is returned if one of the specified columns is not part of the matrix.
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if one of the given columns is greater than the matrix or negative.
+    fn swap_columns(
+        &mut self,
+        col0: impl TryInto<i64> + Display,
+        col1: impl TryInto<i64> + Display,
+    ) -> Result<(), MathError>;
+
+    /// Swaps two rows of the specified matrix.
+    ///
+    /// Parameters:
+    /// - `row0`: specifies the first row which is swapped with the second one
+    /// - `row1`: specifies the second row which is swapped with the first one
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// Otherwise, a [`MathError`] is returned if one of the specified rows is not part of the matrix.
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if one of the given rows is greater than the matrix or negative.
+    fn swap_rows(
+        &mut self,
+        row0: impl TryInto<i64> + Display,
+        row1: impl TryInto<i64> + Display,
+    ) -> Result<(), MathError>;
+}
```

Comment:
I would expect the functionality swap_entries alongside the functions presented by you.
Consider adding it, or adding a corresponding task to the kanboard for discussion

Reply:
Functionality was added, but is to my eye not much nicer than just manually using `get_entry` and `set_entry` in one line